### PR TITLE
Add Rubocop to CI

### PR DESCRIPTION
**Pull request by cjoudrey**
We currently have `.rubocop.yml` but it is not enforced by CI. This means we will end up having code violations in `master` unless every person thinks about running `rubocop` locally, which isn't realistic.

This PR will make CI fail when Rubocop finds bad code.

I also made a few adjustments to our `.robucop.yml` file. We can discuss each change in the diff.

I also ran `rubocop` and corrected offenses in the codebase.

:eyes: @rmosolgo @astorije

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -32,9 +32,6 @@ Metrics/ParameterLists:
 Style/ClassAndModuleChildren:
   EnforcedStyle: nested
 
-Style/DeprecatedHashMethods:
```
Comment (cjoudrey) [inline .rubocop.yml]: This rule does not exist anymore.

File: .rubocop.yml [hunk 2]
```diff
@@ -55,3 +52,18 @@ Style/StringLiterals:
 
 Style/WordArray:
   EnforcedStyle: brackets
+
+Style/StringLiterals:
```
Comment (cjoudrey) [inline .rubocop.yml]: This is double-quotes versus single-quotes. If you want to take a side, we can enable it and fix the entire code base.

Comment (rmosolgo) [inline .rubocop.yml]: :no_good_man: I'm a double-quotes guy but more than that, I'm a don't-hassle-contributors-with-quote-drama guy

**Comment by cjoudrey** [comment]
Same. I just don't think the gains of enabling this rule outweigh the pain.

**Comment by cjoudrey** [comment]
@rmosolgo were you happy with the rest of the changes?

File: .rubocop.yml [hunk 3]
```diff
@@ -55,3 +52,18 @@ Style/StringLiterals:
...
+  Enabled: false
+
+Metrics/ClassLength:
+  Enabled: false
+
+Metrics/ModuleLength:
+  Enabled: false
+
+Metrics/MethodLength:
```
Comment (rmosolgo) [inline .rubocop.yml]: There's `DisabledByDefault: true` above, is that enough for these `Length` cops? (It's ok with me to remove the `ClassLength` cop higher up in the file, too)

File: .rubocop.yml [hunk 4]
```diff
@@ -55,3 +52,18 @@ Style/StringLiterals:
...
+  Enabled: false
+
+Lint/AssignmentInCondition:
+  Enabled: false
```
Comment (rmosolgo) [inline .rubocop.yml]: I'm willing to give this one up (we don't pass on it anyways), but should we remove the previous definition from above ?

Comment (cjoudrey) [inline .rubocop.yml]: @rmosolgo I just realized `Lint/AssignmentInCondition` was already defined above.

We have some code that violates this [like assignment in `while`](https://github.com/rmosolgo/graphql-ruby/blob/2e59be6482ce94a6124ce27766ad4a7910e6c589/lib/graphql/internal_representation/rewrite.rb#L154).

Did you prefer we fix those or disable this rule?

**Comment by rmosolgo** [comment]
I'm no lover of styleguides but I knew this day would come, thanks for getting this going! Sorry I didn't get to it til now, busy day at my _real_ job :P

File: .rubocop.yml (hunk 4 above)
Comment (rmosolgo) [inline .rubocop.yml]: My _personal_ opinion is "yes" for `while` and "no" for `if`, but that doesn't really seem fair to push on people, so turning it off is ok for me!

File: .rubocop.yml (hunk 3 above)
Comment (cjoudrey) [inline .rubocop.yml]: Not sure how I missed that. You're absolutely right. All these `Enabled: false` aren't necessary.

**Comment by cjoudrey** [comment]
> I'm no lover of styleguides but I knew this day would come, thanks for getting this going!

No problem! 😀 

At the very least it'll help us catch bad indentation.

**Comment by rmosolgo** [comment]
:+1:

**Pull request merged**